rllib/utils/serialization.py

```diff
@@ -14,14 +14,34 @@
 from ray.rllib.utils.spaces.repeated import Repeated
 from ray.rllib.utils.spaces.simplex import Simplex
 
+NOT_SERIALIZABLE = "__not_serializable__"
+
 gym, old_gym = try_import_gymnasium_and_gym()
 
 old_gym_text_class = None
 if old_gym:
     old_gym_text_class = getattr(old_gym.spaces, "Text", None)
 
 
-NOT_SERIALIZABLE = "__not_serializable__"
+@DeveloperAPI
+def convert_numpy_to_python_primitives(obj: Any):
+    """Convert an object that is a numpy type to a python type.
+
+    If the object is not a numpy type, it is returned unchanged.
+
+    Args:
+        obj: The object to convert.
+    """
+    if isinstance(obj, np.integer):
+        return int(obj)
+    elif isinstance(obj, np.floating):
+        return float(obj)
+    elif isinstance(obj, np.ndarray):
+        return obj.tolist()
```

Comment:
I think you need to recursively handle this one, since list is not a primitive type.
can you write a unit test for this?

Reply:
ah great catch.

Reply:
in the case where I pass an array that is made up of numpy types this will return a list where the elements are numpy types, which I was trying to avoid to begin with

Reply:
fixed